src/NuGet.Core/NuGet.Protocol.Core.Types/WebRequestEventArgs.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+
+namespace NuGet.Protocol.Core.Types
+{
+    public class WebRequestEventArgs : EventArgs
```

Comment:
Note that I just moved the existing `WebRequestEventArgs` down into a common layer. The Package Manager Console was already using it to log stuff using event handlers.

Created an alternative PR that only affects nuget.exe code without the need for a rather impactful refactoring of the logging code: https://github.com/NuGet/NuGet.Client/issues/138